Documentation/ABI/TestContent.md

```diff
@@ -0,0 +1,178 @@
+|-:|:-:|-|
+| `0x00000000` | &ndash; | Reserved (**do not use**) |
+| `0x74657374` | `'test'` | Test or suite declaration |
+| `0x65786974` | `'exit'` | Exit test |
+
+<!-- When adding cases to this enumeration, be sure to also update the
+corresponding enumeration in TestContentGeneration.swift. -->
+
+#### The version field
+
+This field is currently always `0`. Implementations should ignore structures
+with other version values.
+
+#### The accessor field
+
+The function `accessor` is a C function. When called, it initializes the memory
+at its argument `outValue` to an instance of some Swift type and returns `true`,
+or returns `false` if it could not generate the relevant content. On successful
+return, the caller is responsible for deinitializing the memory at `outValue`
+when done with it.
+
+`accessor` is optional. If it is `nil`, the test content record is ignored. The
+testing library may, in the future, define record kinds that do not provide an
+accessor function (that is, they represent pure compile-time information only.)
+
+The second argument to this function, `hint`, is an optional input that can be
+passed to help the accessor function determine if its corresponding test content
+record matches what the caller is looking for. If the caller passes `nil` as the
+`hint` argument, the accessor behaves as if it matched (that is, no additional
+filtering is performed.)
+
+The concrete Swift type of the value written to `outValue` and the value pointed
+to by `hint` depend on the kind of record:
+
+- For test or suite declarations (kind `0x74657374`), the accessor produces an
+  asynchronous Swift function that returns an instance of `Test`:
+
+  ```swift
+  @Sendable () async -> Test
+  ```
+
+  This signature is not the signature of `accessor`, but of the Swift function
+  reference it writes to `outValue`. This level of indirection is necessary
+  because loading a test or suite declaration is an asynchronous operation, but
+  C functions cannot be `async`.
+
+  Test content records of this kind do not specify a type for `hint`. Always
+  pass `nil`.
+
+- For exit test declarations (kind `0x65786974`), the accessor produces a
+  structure describing the exit test (of type `__ExitTest`.)
+
+  Test content records of this kind accept a `hint` of type `SourceLocation`.
+  They only produce a result if they represent an exit test declared at the same
+  source location (or if the hint is `nil`.)
+
+#### The context field
+
+This field can be used by test content to store additional context for a test
+content record that needs to be made available before the accessor is called:
+
+- For test or suite declarations (kind `0x74657374`), this field contains a bit
+  mask with the following flags currently defined:
+
+  | Bit | Description |
+  |-:|-|
+  | `1 << 0` | This record contains a suite declaration |
+  | `1 << 1` | This record contains a parameterized test function declaration |
```

Comment:
Perhaps consider evaluating the `Bit` column, it makes it easier to follow.

Reply:
e.g. `1 << 1 = 2`?